addons/mail/models/ir_actions_server.py

```diff
@@ -182,41 +193,75 @@ def _compute_followers_info(self):
                 action.followers_partner_field_name = False
 
     @api.depends('model_id', 'state')
-    def _compute_activity_info(self):
-        to_reset = self.filtered(lambda act: not act.model_id or act.state != 'next_activity')
-        if to_reset:
-            to_reset.activity_type_id = False
-            to_reset.activity_summary = False
-            to_reset.activity_note = False
-            to_reset.activity_date_deadline_range = False
-            to_reset.activity_date_deadline_range_type = False
-            to_reset.activity_user_type = False
+    def _compute_has_activity_plans(self):
+        self.has_activity_plans = False
+        for action in self.filtered(lambda act: act.state == 'next_activity'):
+            action.has_activity_plans = self.env['mail.activity.plan'].search_count([
+                '|', ('res_model', '=', False), ('res_model', '=', action.model_id.model)
+            ], limit=1) > 0
+
+    @api.depends('model_id', 'state')
+    def _compute_activity_plan_id(self):
+        to_reset = self.filtered(lambda act: act.state != 'next_activity' or (act.activity_plan_id.res_model and act.model_id.model != act.activity_plan_id.res_model) or act.activity_type_id)
```

Comment:
Not sure why there is a condition about activity_type_id: we are changing model or state, so we should depend on those only. If we have an activity type, we probably don't have a plan anyway.